client/src/components/ConfigurationPanel/ConfigurationPanel.js

```diff
@@ -0,0 +1,85 @@
+import React, { useState } from 'react';
+import {
+    CInputGroup, CInputGroupText, CFormInput, CFormSelect,
+    CButton, CModal, CModalBody, CModalHeader, CModalTitle, CModalFooter,
+    CRow
+} from '@coreui/react';
+import './ConfigurationPanel.css';
+
+const ConfigurationPanel = ({ promptType, setPromptType, numberOfShots, setNumberOfShots, handleGeneratePrompt, generatedPrompt }) => {
+    const [showPromptPreview, setShowPromptPreview] = useState(false);
+
+    const allowedPromptTypes = {
+        "basic": 'Basic',
+        "text_representation": 'Text Representation',
+        "openai_demonstration": 'OpenAI Demonstration',
+        "code_representation": 'Code Representation',
+        "alpaca_sft": 'Alpaca SFT',
+        "full_information": 'Full Information',
+        "sql_only": 'SQL Only',
+        "dail_sql": 'Dail SQL'
+    };
+
+    const handlePreviewClick = async () => {
+        const isPromptGenerated = await handleGeneratePrompt();
+        if (isPromptGenerated && generatedPrompt) {
+            setShowPromptPreview(true);
+        }
+    };
+
+    return (
+        <div className="configuration-panel">
+            <CRow>
+                <h4>Prompt Configuration</h4>
+            </CRow>
+            <CRow>
+                <CInputGroup className="mb-3">
+                    <CInputGroupText as="label" htmlFor="promptType">Prompt Type</CInputGroupText>
+                    <CFormSelect
+                        onChange={(e) => {
+                            setPromptType(e.target.value);
+                            if (!["full_information", "sql_only", "dail_sql"].includes(e.target.value)) {
```

Comment:
These should be constants too but try to accommodate the previous comment in way that it can used here